packages/modules/ad-personalization/src/worker/AdPersonalization.ts

```diff
@@ -0,0 +1,89 @@
+import config from '../data/config.json';
+
+interface SendData {
+  url: string;
+  key: string;
+}
+
+interface Message {
+  action: string;
+  data: SendData;
+}
+
+interface MessageResponse {
+  response: boolean;
+  tabId: number;
+}
+
+const moduleEvents = Object.freeze({
+  AD_PERSONALIZATION: 'ad_personalization',
+});
+
+export class AdPersonalization {
+  private eventEmitter: any;
+
+  constructor() {
+    this.eventEmitter = (self as any).messenger.registerModule('ad-personalization');
```

Comment:
Can we use imported messenger here?

Reply:
I think so, but it will not be consistent with other modules